server/sonar-server/src/main/java/org/sonar/server/usergroups/ws/GroupWsSupport.java

```diff
@@ -0,0 +1,208 @@
+  static final String PARAM_GROUP_NAME = "name";
+  static final String PARAM_GROUP_DESCRIPTION = "description";
+  static final String PARAM_LOGIN = "login";
+
+  // Database column size should be 500 (since migration #353),
+  // but on some instances, column size is still 255,
+  // hence the validation is done with 255
+  static final int DESCRIPTION_MAX_LENGTH = 200;
+
+  private final DbClient dbClient;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
+
+  public GroupWsSupport(DbClient dbClient, DefaultOrganizationProvider defaultOrganizationProvider) {
+    this.dbClient = dbClient;
+    this.defaultOrganizationProvider = defaultOrganizationProvider;
+  }
+
+  /**
+   * Find a group by its id (parameter {@link #PARAM_GROUP_ID}) or couple organization key/group name
+   * (parameters {@link #PARAM_ORGANIZATION_KEY} and {@link #PARAM_GROUP_NAME}). The virtual
+   * group "Anyone" is not supported.
+   *
+   * @throws NotFoundException if parameters are missing/incorrect, if the requested group does not exist
+   * or if the virtual group "Anyone" is requested.
+   */
+  public GroupId findGroup(DbSession dbSession, Request request) {
+    Long id = request.paramAsLong(PARAM_GROUP_ID);
+    String organizationKey = request.param(PARAM_ORGANIZATION_KEY);
+    String name = request.param(PARAM_GROUP_NAME);
+    return findGroup(dbSession, newGroupRef(dbSession, id, organizationKey, name));
+  }
+
+  /**
+   * Finds a user group by its reference. If organization is not defined then group
+   * is searched in default organization.
+   *
+   * @return non-null group
+   * @throws NotFoundException if the requested group does not exist
+   * @throws NotFoundException if the requested group is Anyone
+   */
+  public GroupId findGroup(DbSession dbSession, GroupWsRef ref) {
+    if (ref.hasId()) {
+      GroupDto group = dbClient.groupDao().selectById(dbSession, ref.getId());
+      checkFound(group, "No group with id '%s'", ref.getId());
+      return GroupId.from(group);
+    }
+
+    Optional<GroupDto> group = dbClient.groupDao().selectByName(dbSession, ref.getOrganizationUuid(), ref.getName());
+    checkFoundWithOptional(group, "No group with name '%s'", ref.getName(), ref.getOrganizationUuid());
+    return GroupId.from(group.get());
+  }
+
+  public GroupIdOrAnyone findGroupOrAnyone(DbSession dbSession, GroupWsRef ref) {
+    if (ref.hasId()) {
+      GroupDto group = dbClient.groupDao().selectById(dbSession, ref.getId());
+      checkFound(group, "No group with id '%s'", ref.getId());
+      return GroupIdOrAnyone.from(group);
+    }
+
+    if (ref.isAnyone()) {
+      return GroupIdOrAnyone.forAnyone(ref.getOrganizationUuid());
+    }
+
+    Optional<GroupDto> group = dbClient.groupDao().selectByName(dbSession, ref.getOrganizationUuid(), ref.getName());
+    checkFoundWithOptional(group, "No group with name '%s'", ref.getName(), ref.getOrganizationUuid());
+    return GroupIdOrAnyone.from(group.get());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Call "group.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3655)